web_poet/page_inputs/http.py

```diff
@@ -300,3 +302,14 @@ def _auto_detect_fun(self, body: bytes) -> Optional[str]:
             except UnicodeError:
                 continue
             return resolve_encoding(enc)
+
+
+def request_fingerprint(req: HttpRequest) -> str:
+    """Return the fingerprint of the request."""
+    fp = sha1()
+    fp.update(req.method.encode() + b"\n")
+    fp.update(canonicalize_url(str(req.url)).encode())
```

Comment:
I think we need to add a separator here, so that `request_fingerprint(url+"X", header_name) != request_fingerprint(url, "X"+header_name)` if it makes sense :)